pkg/logs/sender/message_buffer.go

```diff
@@ -14,23 +14,29 @@ type MessageBuffer struct {
 	messageBuffer    []*message.Message
 	contentSize      int
 	contentSizeLimit int
+	bufferSizeLimit  int
 }
 
 // NewMessageBuffer returns a new MessageBuffer.
-func NewMessageBuffer(batchSizeLimit int, contentSizeLimit int) *MessageBuffer {
+func NewMessageBuffer(batchSizeLimit int, contentSizeLimit int, bufferSizeLimit int) *MessageBuffer {
 	return &MessageBuffer{
 		messageBuffer:    make([]*message.Message, 0, batchSizeLimit),
 		contentSizeLimit: contentSizeLimit,
+		bufferSizeLimit:  bufferSizeLimit,
 	}
 }
 
 // AddMessage adds a message to the buffer if there is still some free space,
 // returns true if the message was added.
 func (p *MessageBuffer) AddMessage(message *message.Message) bool {
-	contentSize := len(message.Content)
-	if len(p.messageBuffer) < cap(p.messageBuffer) && p.contentSize+contentSize <= p.contentSizeLimit {
+	newMessageSize := len(message.Content)
+	if newMessageSize > p.contentSizeLimit {
```

Comment:
Can the naming of `contentSizeLimit` be improved to something like `singleMessageSizeLimit`?

Reply:
I changed the internal name in `message_buffer` but left the external name since it's publicly configurable and don't want to introduce a breaking change.